src/site/markdown/index.md

```diff
@@ -19,102 +19,110 @@
 
 ### Maven insights
 
-The challenge of implementing build cache in Maven is that domain model is overly generic and doesn't have dedicated api
-for build inputs. Because of that, even 2 identically looking builds from the same source code could normally produce 2
-different results. The question here is tolerance level - can you accept particular discrepancies or not. For most of
-teams artifacts produced in the same build environment from the same source code will be considered equivalent and
-technical differences between them (like different timestamps in jar manifests) could be ignored. Now consider scenario
-when artifact is first produced with compiler X and cached. Later, without touching source code, compiler changes to Y
-and build yields significantly different outcomes of compilation. Should the produced artifacts be considered as
-equivalent? Both Yes and No answers are possible and could be even desirable in different scenarios. When productivity
-and performance are the primary concerns it could be desirable to tolerate insignificant discrepancies and maximise
-reuse of cached builds. As long as correctness is in focus there could be demand to comply with the exact release
-process. In the same way as with classic Maven, correctness is ensured by proper build configuration and controllable
-build environments. In the same way as with classic Maven the previous build is just an approximation of today build
-with some tolerance (implementation, configuration and environment driven).
+Maven is a proven tool with a long history and core design established many years ago. Historically, Maven's core was
+designed with generic stable interfaces that don't have a concept of inputs and outputs. It just runs as configured, but
+the core does not control the inputs and effects of the run. Most commonly, artifacts produced in the same build
+environment from the same source code will be considered equivalent. But even two identically looking builds from the
+same source code could have two different results. The question here is tolerance level - can you accept particular
+discrepancies? Though technical differences between artifacts like timestamps in manifests are largely ignored, when
+compilers used are of different levels, it is likely a critical difference. Should the produced artifacts be considered
+equivalents? Yes and No answers are possible and could be desirable in different scenarios. When productivity
+and performance are the primary concerns, it could be beneficial to tolerate insignificant discrepancies and maximize
+the reuse. As long as correctness is in focus, there could be a demand to comply with the exact release requirements. In
+the same way as Maven, the cache correctness is ensured by proper build configuration and control over the build
+environment. As Maven itself, the cached result is just an approximation of another build with some tolerance level (
```

Comment:
The opening parenthesis should not be splitted from the next work, so move it on next line.

Reply:
To break the line need to put a double space. In the current form, it just continues. in any case - it moved to the following line